compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -617,6 +620,33 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     });
 
                 if let Some((decl, body_id)) = closure_decl_and_body_id {
+                    let mut visitor = UnresolvedTypeFinder::new(self);
+                    for arg in fn_sig.inputs().skip_binder().iter() {
+                        if let ControlFlow::Break((ty, _)) = visitor.visit_ty(*arg) {
+                            let mut inner = self.inner.borrow_mut();
+                            let ty_vars = &inner.type_variables();
+                            let ty_span = match ty.ty_vid() {
+                                Some(ty_vid) => {
+                                    let var_origin = ty_vars.var_origin(ty_vid);
+                                    var_origin.span
+                                }
+                                None => continue,
+                            };
+                            err.span_label(
+                                ty_span,
+                                "consider giving this closure parameter an explicit type without `_` placeholders",
+                            );
+                            // We don't want to give the other suggestions when the problem is a
+                            // closure input type.
+                            err.span_label(
```

Comment:
Do you think we need to point out the span of the closure? I kinda like the rendered suggestion of `unspecified-parameter.rs`, do you know why it renders differently?

Reply:
Yeah it renders differently because that's the old code path. I'll see if I can change it so they are uniform, I assume it won't be that hard